app/src/main/java/edu/ucsc/cmps115_spring2017/face2name/MainScreen.java

```diff
@@ -104,9 +109,45 @@ public boolean onTouchEvent(MotionEvent event) {
             }
 
         }
+
+        // If we tapped on the screen and if we didn't tap in the text box
+        if(event.getAction() == MotionEvent.ACTION_DOWN && !getLocationOnScreen(nameField).contains(touchX,touchY)) {
+
+            // If the textbox is hidden, bring up the textbox and the keyboard
+            if(nameField.getVisibility() == View.INVISIBLE) {
+                nameField.setVisibility(View.VISIBLE);
+                nameField.requestFocus();
+                InputMethodManager input = (InputMethodManager) getSystemService(MainScreen.INPUT_METHOD_SERVICE);
+                input.showSoftInput(nameField, InputMethodManager.SHOW_IMPLICIT);
+            }
+
+            // Otherwise, hide the textbox and close the keyboard
+            else {
+                InputMethodManager input = (InputMethodManager) getSystemService(MainScreen.INPUT_METHOD_SERVICE);
+                input.hideSoftInputFromWindow(nameField.getWindowToken(), 0);
+                nameField.setVisibility(View.INVISIBLE);
+                nameField.clearFocus();
+            }
+        }
         return true;
     }
 
+    // Makes a rectangle so we can check if we tapped inside of our textbox
+    protected Rect getLocationOnScreen(EditText mEditText) {
```

Comment:
There is no need to call `getLocatonOnScreen()` outside of `MainScreen`'s definition. Declaring the method as `private` would be more suitable.